runtime/runtime/src/lib.rs

```diff
@@ -661,7 +668,9 @@ impl Runtime {
                 validator_reward -= receiver_reward;
                 account.amount = safe_add_balance(account.amount, receiver_reward)?;
                 set_account(state_update, account_id, account);
-                state_update.commit();
+                state_update.commit(StateChangeCause::ReceiptProcessing {
```

Comment:
Please change this `Cause` to `ActionReceiptGasReward`. It's only happen when the account receives a kickback for burnt gas from the contract execution.